picard/util/pipe.py

```diff
@@ -0,0 +1,218 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright (C) 2022 skelly37
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+
+import concurrent.futures
+import os
+from sys import platform
+
+
+IS_WIN: bool = False
+
+if platform == "win32" or platform == "cygwin":
```

Comment:
```suggestion
if platform in {"win32", "cygwin"}:
```

Reply:
BTW, I think we can just remove this (re-) definition of `IS_WIN ` and modify https://github.com/metabrainz/picard/blob/04ddb9f3f076b7940fd6325e271a21a851c3ae44/picard/const/sys.py#L26

@phw is there any reason to not match `cygwin` in `IS_WIN` definition?

Reply:
Okay, so I'll import `IS_WIN` constant and if `cygwin` needs to be supported, we will need to change the constant anyway

Reply:
> @phw is there any reason to not match `cygwin` in `IS_WIN` definition?

There might be, that needs to be investigated per use. As cygwin provides a Unix like environment on Windows it is somewhere in between being Windows and being Unix and not everything might work the same (e.g. file paths) and not every `IS_WIN` usage might apply.

In general this is something not many have probably tried so far. I haven't seen any issue reports specific to cygwin. That either means things work well enough or not enough people have tried it yet :)

Reply:
The discussion is resolved by `from picard.const.sys import IS_WIN` imo.

At least for now because win32 vs cygwin applies to the whole Picard app and its design, not just my PR. I'll just stick to the global way of detecting Windows